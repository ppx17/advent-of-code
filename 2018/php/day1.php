<?php
$InputFile = "../input/input-day1.txt";

$deltas = array_map("intval", explode("\n", trim(file_get_contents($InputFile))));

echo "Part 1: ".array_sum($deltas).PHP_EOL;

$frequency = 0;
$seen = [];

while(true) {
    foreach($deltas as $statement) {
        $frequency += $statement;
        if(isset($seen[$frequency])) {
            echo "Part 2: ".$frequency.PHP_EOL;
            break 2;
        }
        $seen[$frequency] = true;
    }
}
