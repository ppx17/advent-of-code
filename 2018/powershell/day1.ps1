Param(
    [string]$InputFile = '../input/input-day1.txt'
);

$Statements = (Get-Content $InputFile) | Foreach-Object { [Int32]::Parse($_) };

$Sum = 0;
$Part1 = $Part2 = $null;
$History = [System.Collections.ArrayList]::new();
while ($null -eq $Part2) {
    $Statements | Foreach-Object { 
        $Sum += $_; 
        if ($null -eq $Part2 -and $History.IndexOf($Sum) -ge 0) {
            $Part2 = $Sum;
        }
        [void]$History.Add($Sum);
    };
    if ($null -eq $Part1) {
        $Part1 = $Sum;
    }
}

Write-Output "Part 1: ${Part1}";
Write-Output "Part 2: ${Part2}";