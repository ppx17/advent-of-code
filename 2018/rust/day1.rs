use std::fs;

fn main() {
    let filename = "../input/input-day1.txt";
    let contents = fs::read_to_string(filename)
        .unwrap()
        .lines()
        .map(|s| s.parse::<i32>().unwrap());

    println!("Content: \n{}", contents)
}