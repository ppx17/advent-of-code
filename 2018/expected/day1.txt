Part 1: 561
Part 2: 563
